In [1]:
import pandas as pd
import numpy as  np

In [2]:
Data=pd.read_csv("D:\LetsUpgrade\Day 16 & 17\\titles.csv")

In [3]:
Data.head()

title  year
0    Tasveer Mere Sanam  1996
1              Only You  1994
2  El pueblo del terror  1970
3               Machine  2007
4                  MARy  2008

In [4]:
Data.tail()

title  year
244909  Black Butterfly in a Colorful World  2018
244910        Hua fei hua wu chun man cheng  1980
244911                 Nippon dabi katsukyu  1970
244912        Under Siege 2: Dark Territory  1995
244913            She Must Be Seeing Things  1987

In [5]:
Data.shape

(244914, 2)

In [6]:
len(Data)

244914

## 1. How many movies are listed in the titles dataframe?

___There are Total 244914 movies are listed in the dataFrame___

## 2. What are the earliest two films listed in the titles dataframe?

In [7]:
Data.sort_values("year").head(2)

title  year
193222          Miss Jerry  1894
51388   The Startled Lover  1898

## 3. How many movies have the title "Hamlet"?

In [8]:
len(Data[Data.title=='Hamlet'])

20

## 4. How many movies are titled "North by Northwest"?

In [9]:
len(Data[Data.title=='North by Northwest'])

1

## 5. When was the first movie titled "Hamlet" made?

In [10]:
Data[Data.title=='Hamlet'].sort_values("year").head(1)

title  year
52521  Hamlet  1910

## 6. List all of the "Treasure Island" movies from earliest to most recent.

In [11]:
Data[Data.title=='Treasure Island'].sort_values("year")

title  year
55686   Treasure Island  1918
196514  Treasure Island  1920
141822  Treasure Island  1934
77354   Treasure Island  1950
65051   Treasure Island  1972
37238   Treasure Island  1973
173051  Treasure Island  1985
27076   Treasure Island  1999

## 7. How many movies were made in the year 1950?

In [12]:
len(Data[Data.year==1950])

1113

## 8. How many movies were made in the year 1960?

In [13]:
len(Data[Data.year==1960])

1549

## 9. How many movies were made from 1950 through 1959?

In [14]:
len(Data[(1950<=Data.year) & (Data.year<=1959)])

12934

## 10. In what years has a movie titled "Batman" been released?

In [15]:
Data[Data.title=='Batman']

title  year
39349   Batman  1943
240896  Batman  1989

In [16]:
Cast=pd.read_csv("D:\LetsUpgrade\Day 16 & 17\\cast.csv")

In [17]:
Cast.head()

title  year      name   type                character     n
0        Closet Monster  2015  Buffy #1  actor                  Buffy 4  31.0
1       Suuri illusioni  1985    Homo $  actor                   Guests  22.0
2   Battle of the Sexes  2017   $hutter  actor          Bobby Riggs Fan  10.0
3  Secret in Their Eyes  2015   $hutter  actor          2002 Dodger Fan   NaN
4            Steve Jobs  2015   $hutter  actor  1988 Opera House Patron   NaN

In [18]:
Cast.tail()

title  year                         name  \
3786171                           Foxtrot  1988            Lilja ??risd?ttir   
3786172  Niceland (Population. 1.000.002)  2004    Sigr??ur J?na ??risd?ttir   
3786173                         Skammdegi  1985           Dalla ??r?ard?ttir   
3786174                      U.S.S.S.S...  2003  Krist?n Andrea ??r?ard?ttir   
3786175                 Bye Bye Blue Bird  1999                 Rosa ? R?gvu   

            type               character     n  
3786171  actress                    D?ra  24.0  
3786172  actress            Woman in Bus  26.0  
3786173  actress           Hj?krunarkona   9.0  
3786174  actress  Afgr.dama ? bens?nst??  17.0  
3786175  actress       Pensionatv?rtinde   NaN

## 11. How many roles were there in the movie "Inception"?

In [19]:
len(Cast[Cast.title=="Inception"])

78

## 12. How many roles in the movie "Inception" are NOT ranked by an "n" value?

In [20]:
c=Cast
c=c[c.title=="Inception"]
c=c[c.n.isnull()]
len(c)

27

## 13. But how many roles in the movie "Inception" did receive an "n" value?¶


In [21]:
c=Cast
c=c[c.title=="Inception"]
c=c[c.n.notnull()]
len(c)

51

## 14. Display the cast of "North by Northwest" in their correct "n"-value order, ignoring roles that did not earn a numeric "n" value.

In [22]:
c =Cast
c=Cast[Cast.title=='North by Northwest'].sort_values("n")
c=c[c.n.notnull()]
c

title  year                  name     type  \
871426   North by Northwest  1959            Cary Grant    actor   
3518333  North by Northwest  1959       Eva Marie Saint  actress   
1457165  North by Northwest  1959           James Mason    actor   
3157232  North by Northwest  1959   Jessie Royce Landis  actress   
356707   North by Northwest  1959        Leo G. Carroll    actor   
3048272  North by Northwest  1959  Josephine Hutchinson  actress   
1695173  North by Northwest  1959           Philip Ober    actor   
1275337  North by Northwest  1959         Martin Landau    actor   
2447081  North by Northwest  1959         Adam Williams    actor   
1811387  North by Northwest  1959          Edward Platt    actor   
665768   North by Northwest  1959     Robert Ellenstein    actor   
2295714  North by Northwest  1959          Les Tremayne    actor   
464736   North by Northwest  1959       Philip Coolidge    actor   
1508344  North by Northwest  1959         Patrick McVey    actor   
206471   North by Northwest  1959          Edward Binns    actor   
1384780  North by Northwest  1959             Ken Lynch    actor   

                           character     n  
871426            Roger O. Thornhill   1.0  
3518333                  Eve Kendall   2.0  
1457165              Phillip Vandamm   3.0  
3157232              Clara Thornhill   4.0  
356707                 The Professor   5.0  
3048272                Mrs. Townsend   6.0  
1695173              Lester Townsend   7.0  
1275337                      Leonard   8.0  
2447081                     Valerian   9.0  
1811387              Victor Larrabee  10.0  
665768                         Licht  11.0  
2295714                   Auctioneer  12.0  
464736                     Dr. Cross  13.0  
1508344               Sergeant Flamm  14.0  
206471                Captain Junket  15.0  
1384780  Charley - Chicago Policeman  16.0

## 15. Display the entire cast, in "n"-order, of the 1972 film "Sleuth"

In [23]:
Cast[(Cast["title"]=="Sleuth") & (Cast["year"]==1972)]

title  year                name     type  \
326773   Sleuth  1972       Michael Caine    actor   
373876   Sleuth  1972      Alec Cawthorne    actor   
1448690  Sleuth  1972        Teddy Martin    actor   
1465483  Sleuth  1972  John (II) Matthews    actor   
1705394  Sleuth  1972    Laurence Olivier    actor   
2724553  Sleuth  1972  Eve (III) Channing  actress   

                          character    n  
326773                  Milo Tindle  2.0  
373876            Inspector Doppler  3.0  
1448690      Police Constable Higgs  6.0  
1465483  Detective Sergeant Tarrant  4.0  
1705394                 Andrew Wyke  1.0  
2724553             Marguerite Wyke  5.0

## 16. Now display the entire cast, in "n"-order, of the 2007 version of "Sleuth".

In [24]:
Cast[(Cast["title"]=="Sleuth") & (Cast["year"]==2007)]

title  year                 name     type          character    n
260242   Sleuth  2007      Kenneth Branagh    actor  Other Man on T.V.  NaN
326774   Sleuth  2007        Michael Caine    actor             Andrew  1.0
373877   Sleuth  2007  Alec (II) Cawthorne    actor  Inspector Doppler  NaN
1293634  Sleuth  2007             Jude Law    actor               Milo  2.0
1805907  Sleuth  2007        Harold Pinter    actor        Man on T.V.  3.0
2724552  Sleuth  2007    Eve (II) Channing  actress    Marguerite Wyke  NaN
3370424  Sleuth  2007    Carmel O'Sullivan  actress             Maggie  NaN

In [25]:
Cast[Cast.title=='Sleuth'].sort_values('n')

title  year                 name     type  \
326774   Sleuth  2007        Michael Caine    actor   
1705394  Sleuth  1972     Laurence Olivier    actor   
326773   Sleuth  1972        Michael Caine    actor   
1293634  Sleuth  2007             Jude Law    actor   
373876   Sleuth  1972       Alec Cawthorne    actor   
1805907  Sleuth  2007        Harold Pinter    actor   
1465483  Sleuth  1972   John (II) Matthews    actor   
2724553  Sleuth  1972   Eve (III) Channing  actress   
1448690  Sleuth  1972         Teddy Martin    actor   
260242   Sleuth  2007      Kenneth Branagh    actor   
373877   Sleuth  2007  Alec (II) Cawthorne    actor   
2724552  Sleuth  2007    Eve (II) Channing  actress   
3370424  Sleuth  2007    Carmel O'Sullivan  actress   

                          character    n  
326774                       Andrew  1.0  
1705394                 Andrew Wyke  1.0  
326773                  Milo Tindle  2.0  
1293634                        Milo  2.0  
373876            Inspector Doppler  3.0  
1805907                 Man on T.V.  3.0  
1465483  Detective Sergeant Tarrant  4.0  
2724553             Marguerite Wyke  5.0  
1448690      Police Constable Higgs  6.0  
260242            Other Man on T.V.  NaN  
373877            Inspector Doppler  NaN  
2724552             Marguerite Wyke  NaN  
3370424                      Maggie  NaN

## 17. How many roles were credited in the silent 1921 version of Hamlet?

In [26]:
Cast[(Cast["title"]=="Hamlet") & (Cast["year"]==1921)]

title  year                    name     type         character    n
7789     Hamlet  1921        Fritz Achterberg    actor       Fortinbras,  9.0
458904   Hamlet  1921            Paul Conradi    actor      K?nig Hamlet  2.0
555034   Hamlet  1921        Anton De Verdier    actor           Laertes  7.0
1135279  Hamlet  1921         Hans Junkermann    actor          Polonius  6.0
2185558  Hamlet  1921            Heinz Stieda    actor           Horatio  5.0
2375993  Hamlet  1921  Eduard von Winterstein    actor          Claudius  4.0
2664423  Hamlet  1921         Mathilde Brandt  actress  K?nigin Gertrude  3.0
3062023  Hamlet  1921          Lilly Jacobson  actress           Ophelia  8.0
3355176  Hamlet  1921            Asta Nielsen  actress            Hamlet  1.0

## 18. How many roles were credited in Branagh’s 1996 Hamlet?

In [27]:
Cast[(Cast["title"]=="Hamlet") & (Cast["year"]==1996)].character.shape[0]

54

## 19. How many "Hamlet" roles have been listed in all film credits through history?

In [28]:
len(Cast[(Cast["title"]=="Hamlet")])

342

## 20. How many people have played an "Ophelia"?

In [29]:
Cast[(Cast["title"]=="Ophelia")].character.shape[0]

20

## 21. How many people have played a role called "The Dude"?

In [30]:
Cast[(Cast["title"]=="The Dude")].character.shape[0]

0

## 22.How many people have played a role called "The Stranger"?

In [31]:
Cast[(Cast["title"]=="The Stranger")].character.shape[0]

106

## 23. How many roles has Sidney Poitier played throughout his career?

In [32]:
Cast[(Cast["character"]=="Sidney Poitier")].character.shape[0]

1

## 24. How many roles has Judi Dench played?

In [33]:
Cast[(Cast["character"]=="Judi Dench")].character.shape[0]

0

## 25. List the supporting roles (having n=2) played by Cary Grant in the 1940s, in order by year.

In [34]:
len(Cast[(Cast["character"]=="Cary Grant") & (Cast["n"]==2)])

1

## 26. List the leading roles that Cary Grant played in the 1940s in order by year.

In [35]:
Cast[(Cast["character"]=="Cary Grant") & (Cast["year"]==1940)].character.shape[0]

0

## 27. How many roles were available for actors in the 1950s?

In [36]:
Cast[(Cast["type"]=="actors") & (Cast["year"]==1950)].character.shape[0]

0

## 28. How many roles were available for actresses in the 1950s?

In [37]:
Cast[(Cast["type"]=="actresses") & (Cast["year"]==1950)].character.shape[0]

0

## 29. How many leading roles (n=1) were available from the beginning of film history through 1980? 

In [38]:
len(Cast[(Cast["year"]=="1980") & (Cast["n"]==1)])

C:\Users\Ash\Anaconda3\lib\site-packages\pandas\core\ops\array_ops.py:253: FutureWarning: elementwise comparison failed; returning scalar instead, but in the future will perform elementwise comparison
  res_values = method(rvalues)


0

## 30. How many non-leading roles were available through from the beginning of film history through 1980?

In [39]:
len(Cast[(Cast["year"]=="1980")])

0

## 31. How many roles through 1980 were minor enough that they did not warrant a numeric "n" rank?

In [40]:
len(Cast[(Cast["year"]=="1980") & (Cast["n"]=='NaN')])

0